src/client/datascience/jupyter/kernels/kernelConnectionWrapper.ts

```diff
@@ -128,13 +128,13 @@ export class KernelConnectionWrapper implements Kernel.IKernelConnection {
         return this.kernelConnection?.connectionStatus || 'disconnected';
     }
     public get info(): Promise<IInfoReply> {
-        return (this.kernelConnection || this._previousKernelConnection).info;
+        return (this.kernelConnection || this._initialKernelConnection).info;
```

Comment:
It seems weird to me to make sure these don't crash but that they might return incorrect information. The _initialKernelConnection may be long dead when the caller makes this request.

Reply:
Is there a way we can make the 3rd party api more robust/obvious? Maybe just have it return undefined when the restart is going on until it's back up?

Reply:
> _initialKernelConnection may be long dead when the caller makes this request.

Yes, but the info never changes, hence I'm retruning a cached info. The info is basically a kernel informaiton (version etc).

> Maybe just have it return undefined when the restart is going on until it's back up?

No, based on the typings it cannot be undefined. Hence the reason to cache it.

Reply:
I think we should discuss this at standup.